docs/api-reference/communicatorlib.md

```diff
@@ -72,3 +72,69 @@ This communicator is provided in the `@dbos-inc/communicator-random` package.
 
 ### `random()`
 `random` is a wrapper for `Math.random()` and similarly produces a `number` in the range from 0 to 1.
+
+## Amazon Web Services (AWS) Communicators
+
+### AWS Configuration
+Configuration of AWS services typically relies on access keys, which are needed by the application to make service API calls, but also are to be kept secret.
+
+There are multiple strategies used in AWS deployments, including the following:
+- Use one access key for the whole application; this generally is the key for an IAM role that is authorized to use all APIs that the application requires
+- Use one access key for each service used by the application
+- Use multiple access keys for each service, potentially with different permissions for roles within the application
+
+DBOS Transact is designed to make configuration with a single access key straightforward, while allowing more flexible configurations.
+
+First, AWS configuration sections are added to the `application` part of `dbos-config.yaml`:
+```yaml
+application:
+  aws_config:
+    aws_region: ${AWS_REGION}
+    aws_access_key_id: ${AWS_ACCESS_KEY_ID}
+    aws_secret_access_key: ${AWS_SECRET_ACCESS_KEY}
+```
+
+The default configuration section of `application.aws_config` is used for any communicator that has not been specifically configured.
+
+Individual AWS services can override this, for example the SES communicator uses `aws_ses_configuration` to specify the configuration(s) for use by SES:
```

Comment:
At least to me, all these defaults are a bit confusing and un-discoverable (how would I know it's called `aws_ses_configuration`?), especially as all the SES examples specify a particular named configuration.

Reply:
Yes, so none of these are discoverable.  aws_config, aws_config.aws_region, aws_config.aws_access_key_id, aws_config.aws_secret_access_key, aws_ses_configuration, not discoverable and not error checked; you could potentially "discover" them from the README or cut+paste an example.
What is the proposal to make them discoverable?  That a library has a way to extend the application schema section for the config file in some way?

Reply:
I added another example in the README.md for SES (in the S3 PR), which is one place it belongs.  I will do something here too.

Reply:
My proposal is to get rid of them (because they're undiscoverable) and require the name always be supplied in `configureInstance`.